Dockerfile

```diff
@@ -1,9 +1,46 @@
+# docker build --force-rm --squash -t {REPO}/{NAME}:{TAG} .
+# --squash requires the expermimental flag to be set.
+# https://docs.docker.com/engine/reference/commandline/dockerd/#description
+#
+# This is based on the original Golang Dockerfile for Debian Stretch
+# https://github.com/docker-library/golang/blob/906e04de73168f643c5c2b40dca0877a14d2377c/1.10/stretch/Dockerfile
+
 FROM golang:latest
 MAINTAINER Julien Vehent
-COPY . $GOPATH/src/github.com/mozilla/tls-observatory
 
-RUN addgroup -gid 10001 app && \
+# install go1.11
+RUN	git clone https://github.com/golang/go.git /tmp/go; \
+	cd /tmp/go/src; \
+	git checkout 4458a357ab819a612c0c4cafae88a65287254fe9; \
+	./make.bash; \
+	rm -rf /usr/local/go; \
+	mv /tmp/go /usr/local/; \
+	rm -rf /usr/local/go/.git*; \
+	rm -rf /tmp/*; \
+	go version
+
+ENV GOPATH /go
+ENV PATH $GOPATH/bin:/usr/local/go/bin:$PATH
+
+RUN mkdir -p "$GOPATH/src" "$GOPATH/bin" && chmod -R 777 "$GOPATH"
+WORKDIR $GOPATH
+
+COPY . $GOPATH/src/github.com/mozilla/tls-observatory
+RUN rm -rf $GOPATH/src/github.com/mozilla/tls-observatory/.git && \
+    # Build the latest Go from master
+    git clone https://github.com/golang/go.git /tmp/go && \
+	cd /tmp/go/src && \
+	#git checkout 4458a357ab819a612c0c4cafae88a65287254fe9 && \
+	./make.bash && \
```

Comment:
Why are we building Go twice?

Reply:
iterating. wait a bit before reviewing ;)